1140, Replace QTip2 with Hint.css

chahmedejaz opened the pull request:
Hi there!
This PR replaces the QTips2 package functionality with the Hint.css as per issue #1140. 
**Here are the results after the replacement:**
![image](https://github.com/github/choosealicense.com/assets/59338032/1e8120b4-3cbc-4f1c-90a8-90bba2281e13)

![image](https://github.com/github/choosealicense.com/assets/59338032/8e4ec014-9bf2-46e8-9481-3c2f56401b9c)

**Here's the point to be noted:**
Previously we were showing the formatted error message for the repository URL input field error:
![image](https://github.com/github/choosealicense.com/assets/59338032/520d9253-c7be-4011-940b-24fcce557273)

However, the hint.css package doesn't support the html tags in the tooltip body. Due to this limitation, we are now showing the error message in plain text:
![image](https://github.com/github/choosealicense.com/assets/59338032/a49a8276-3772-4dc8-ae85-b4099ef3c8f1)

chahmedejaz committed:
1140, add hint.css

chahmedejaz committed:
#1140, update hint.css for licenses

chahmedejaz committed:
#1140, update the font size for tooltip's bodies

chahmedejaz committed:
#1140, update hint.css for repository url input field errors

chahmedejaz committed:
#1140, uninstall qtip2

mlinksva commented:
Thanks @chahmedejaz! I tried it out and have a few comments.

First, something changed with the /appendix page; the green/blue/red circles are now flush to the left of their table cells, rather than centered:

![image](https://github.com/github/choosealicense.com/assets/40415/42670251-7505-4ec8-8a0a-b1be639bb6e9)

Second, in a very narrow browser window, the tooltips get cut off; this is not a huge deal as it was the case for the existing site until very recently https://github.com/github/choosealicense.com/pull/1133 but it'd be nice not to regress:

![Screenshot 2023-08-28 at 6 29 47 PM](https://github.com/github/choosealicense.com/assets/40415/6f0bcbaf-3029-44d6-b5af-f3e9880dbd18)

Third and least importantly, light text on darker color background (taken from the current heading?) strikes me as harder to read than darker color text on light color background (like the current message). Added: I might be wrong, guess it's a matter of taste.

chahmedejaz commented:
Thank you @mlinksva for the review. Let me address the comments and get back to you :smile:

chahmedejaz committed:
114, center the appendix table sprites

chahmedejaz committed:
1140, update tooltips background colors

chahmedejaz committed:
1140, make tooltips responsive

chahmedejaz commented:
Hey @mlinksva - I've addressed your comments. Please review the fixes.
- [first comment](https://github.com/github/choosealicense.com/pull/1141/commits/94ec385ac93f3e383045645e4b41b70ccc883fb9)
- [second comment](https://github.com/github/choosealicense.com/pull/1141/commits/d4b0fadeb245b6e566d359a435cddf176ea90da6)
- [third comment](https://github.com/github/choosealicense.com/pull/1141/commits/5c0df8383145d9481f192d3bcbe64a727787693d)

**Results:**
Now the tooltip is responsive:
![image](https://github.com/github/choosealicense.com/assets/59338032/bd8a10cc-7f25-4e0c-8502-d73416a8dc3b)

Tooltip colors:
![image](https://github.com/github/choosealicense.com/assets/59338032/abd0152b-9893-4c8a-86ba-f3fc0868e5a8)
![image](https://github.com/github/choosealicense.com/assets/59338032/7f06eaca-2b17-40f5-baa1-b4b4b110d572)
![image](https://github.com/github/choosealicense.com/assets/59338032/75debb87-9f28-4c37-9b17-146687de032e)

The appendix page is as per the original site now:
![image](https://github.com/github/choosealicense.com/assets/59338032/d9e12e90-9876-4559-8faf-12bc10711a9a)

Please let me know if I'm missing anything. Thanks.

mlinksva commented:
The screenshots look good, but when I run it, the tooltips have odd shadow or doubled text:

<img width="840" alt="Screenshot 2023-08-30 at 5 19 41 PM" src="https://github.com/github/choosealicense.com/assets/40415/5ed9e546-2854-4928-880c-98e08d22d218">

This is on a Mac FWIW, but I get the same on both Firefox and Chrome.

Any idea what could be causing that?

mlinksva commented on assets/js/app.coffee:
```diff
@@ -16,18 +16,17 @@ class Choosealicense
-    limitations: "Limitation"
+  tooltipAttributesMapperByRuleType:
+    permissions:
+      heading: 'Permissions'
```
```suggestion
      heading: 'Permission'
```

This should be singular, same for the two below.

chahmedejaz replied:
@mlinksva  - Updated [here](https://github.com/github/choosealicense.com/pull/1141/commits/591dd6a388bbb05953dddce6979eac45af9b5b2b).

mlinksva commented on assets/js/app.coffee:
```diff
@@ -16,18 +16,17 @@ class Choosealicense
+  tooltipAttributesMapperByRuleType:
+    permissions:
+      heading: 'Permissions'
+      color: 'hint--success'
```
It looks like success/info/error are built into hint.css and while the colors basically match what's needed, I'd probably prefer to separately define permission/condition/limitation colors as success/info/error aren't semantically correct, even if it is just code the end user won't see.

chahmedejaz replied:
@mlinksva  - Updated [here](https://github.com/github/choosealicense.com/pull/1141/commits/c71d2749bab23b306550b1066aa0541a81031c1a).

chahmedejaz committed:
1140, update tooltip headings

chahmedejaz committed:
1140, update tooltip color class names

chahmedejaz committed:
1140, fix tooltip body text issue

chahmedejaz commented:
@mlinksva  - 

- Actually this effect is from the hint.css
- It is putting the shadow on the tool-tip body text and causing the doubled-text effect
- I've [overridden this shadow to none](https://github.com/github/choosealicense.com/pull/1141/commits/56122d32885e833c66702b3016e6f7700c126747) for our use and hopefully it will do the fix
- It's putting the shadow at my end as well but apparently it's more prominent at your end :sweat_smile: 

**Result:**
![image](https://github.com/github/choosealicense.com/assets/59338032/ac6a62a4-d6c9-4085-b3d6-58020b555019)

chahmedejaz commented:
Hey @mlinksva - I've addressed your comments, please review. Thanks.

chahmedejaz committed:
1140, Update the tooltip arrow color

chahmedejaz commented:
I noticed that the tooltip arrow colors were lost due to the new color class names. Updated the arrow color for the new classes [here](https://github.com/github/choosealicense.com/pull/1141/commits/0987bb0156a6144608de8fc99d5478d31f0c1c02)

chahmedejaz committed:
1140, Update the color for repo input field tooltip error

chahmedejaz committed:
1140, update the tooltip arrow color for the repo input field

mlinksva committed:
Merge branch 'gh-pages' into 1140-replace-qtip2-with-hint

mlinksva commented on .gitignore:
```diff
@@ -5,7 +5,6 @@
 assets/vendor/clipboard/src
 assets/vendor/clipboard/test
 assets/vendor/selectivizr/tests
-assets/vendor/qtip2/basic
```
Seems like we could ignore hint.css/src?

chahmedejaz replied:
Yes @mlinksva , we can. Sorry, It skipped my mind before :sweat_smile: 
Updated and removed the ignored source files from the branch as well.

mlinksva reviewed (changes requested):
Excellent, all looks like it's working. One final comment: think we could avoid vendoring hint.css source

chahmedejaz committed:
1140, add hint.css source in .gitignore

chahmedejaz committed:
1140, delete hint.css source files

mlinksva reviewed (approved):
🎉 @chahmedejaz, well done and thanks
